### PR TITLE
dev-util/bloaty: bump to 1.1, add live ebuild, add myself as proxy-maintainer

### PR DESCRIPTION
**Pull request by ngg**
Closes: https://bugs.gentoo.org/717124
Closes: https://bugs.gentoo.org/724160

### CODE REVIEW

File: dev-util/bloaty/bloaty-1.1.ebuild [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+EAPI=7
+
+inherit cmake
+
+DESCRIPTION="A size profiler for binaries"
+HOMEPAGE="https://github.com/google/bloaty"
+LICENSE="Apache-2.0"
+SLOT="0"
+IUSE=""
```
Comment (juippis) [inline dev-util/bloaty/bloaty-1.1.ebuild:12]: Empty variable can be dropped from the ebuild.

File: dev-util/bloaty/bloaty-1.1.ebuild [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+		-DBLOATY_ENABLE_CMAKETARGETS=OFF
+		-DBUILD_SHARED_LIBS=OFF
+		-DCMAKE_SKIP_RPATH=ON
+	)
+	if [[ ${PV} == 9999 ]]; then
+		mycmakeargs+=(
+			-DBUILD_TESTING=$(usex test)
+			$(usex test -DINSTALL_GTEST=OFF "")
+		)
+	fi
```
Comment (juippis) [inline dev-util/bloaty/bloaty-1.1.ebuild:45]: Are you sure these are not enabled on 1.1 release? Didn't download tarball but by browsing files on release tag, they seem to be present.

Comment (ngg) [reply dev-util/bloaty/bloaty-1.1.ebuild:45]: Tests are not part of the release tarball, they can only be run when cloning from git, that's why it's only enabled in 9999. Regarding the gtest instalation issue, I sent a PR to fix it upstream, but it was only merged after the 1.1 release, that's why the INSTALL_GTEST=OFF is necessary here.

File: dev-util/bloaty/bloaty-1.1.ebuild [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+	dev-libs/protobuf:=
+	dev-libs/re2:=
+"
+RDEPEND="${DEPEND}"
+
+src_configure() {
+	local mycmakeargs=(
+		-DBLOATY_ENABLE_CMAKETARGETS=OFF
+		-DBUILD_SHARED_LIBS=OFF
+		-DCMAKE_SKIP_RPATH=ON
```
[the remark this answers is not in the document]
Comment (ngg) [reply dev-util/bloaty/bloaty-1.1.ebuild:38]: If building without BUILD_SHARED_LIBS=OFF cmake creates a `liblibbloaty.so` and `bloaty` uses it. It's not designed to be used as a library, so linking it statically into the binary is easier. Actually now I see that it has been fixed on the master branch to force it to be a static library, I'll update the 9999 version.

[the remark this answers is not in the document]
Comment (ngg) [reply dev-util/bloaty/bloaty-1.1.ebuild:38]: I'll check if CMAKE_SKIP_RPATH is really required

[the remark this answers is not in the document]
Comment (ngg) [reply dev-util/bloaty/bloaty-1.1.ebuild:38]: It seems to me that's not required, I'll remove.

[the remark this answers is not in the document]
Comment (ngg) [reply dev-util/bloaty/bloaty-1.1.ebuild:38]: Without BLOATY_ENABLE_CMAKETARGETS=OFF there will be cmake target files installed to /usr/lib/Bloaty. As bloaty is a single binary file, I don't think those cmake files are of much use to anyone, but if you think they should be installed anyway, let me know.

**Pull request closed without merging**